client/scripts/views/pages/search.ts

```diff
@@ -153,41 +172,36 @@ export const getDiscussionResult = (thread, searchTerm) => {
               });
             }
           })(),
-        ]),
+        ])
       ]
-    ]),
-    contentLeft: m(ReplyIcon)
+    ),
+    contentLeft: m(ReplyIcon),
   });
 };
 
-const getListing = (results: any, searchTerm: string, pageCount: number, searchType?: SearchType) => {
-  if (Object.keys(results).length === 0) return [];
-  const filter = searchType === SearchType.Top ? null : searchType;
-  const concatResults = () => {
-    let allResults = [];
-    [SearchType.Discussion, SearchType.Member, SearchType.Community].forEach((type) => {
-      if (results[type]?.length) {
-        allResults = allResults.concat(results[type]);
-      }
-    });
-    return allResults;
-  };
-  const tabScopedResults = (filter ? results[searchType] : concatResults())
+const getListing = (results: any, searchTerm: string, pageCount: number, sort: SearchSort, searchType?: SearchScope) => {
+  if (Object.keys(results).length === 0 || !results[searchType]) return [];
+  const tabScopedResults = (results[searchType])
     .sort((a, b) => {
-      // TODO: Token-sorting approach
-      // Some users are not verified; we give them a default date of 1900
-      const aCreatedAt = moment(a.created_at || a.createdAt || a.verified || '1900-01-01T:00:00:00Z');
-      const bCreatedAt = moment(b.created_at || b.createdAt || b.verified || '1900-01-01T:00:00:00Z');
-      return bCreatedAt.diff(aCreatedAt);
+      if (sort === SearchSort.Best && (searchType === SearchScope.Threads || searchType === SearchScope.Replies)) {
+        return b.rank - a.rank
```

Comment:
In addition to ts_rank's lexical/formal ranking, would be great to eventually take into account whether a result is in title or body of a discussion, and how much engagement a post has seen, but we can kick the can down the line there—let's just make sure to push the ["pt 3" of PR 1 ](https://www.notion.so/hicommonwealth/Search-Upgrade-620316a5713f445286579c0a00dcc6a8#ba480db086b24b7a945c670febdaeee5) to a future PR so it isn't forgotten.

Reply:
So far the things we take into account when calculating rank is only the title vs body, but in the future we could do this better two way: 1) use an actual document search technology (Elasticsearch/Sphinx/Lucene) or 2) incorporate more metrics on our own. Option 2 is possible but requires a surprisingly deep understanding of the data science used in search ranking - way out of scope for me. I'm thinking either option could be explored in the future though, so I'll add it in the documentation.

Reply:
Also to note, I took this actual line out of the PR since the ranking is actually all being done on the backend and there's no reason to re-sort here.